src/actions.js

```diff
@@ -143,6 +144,19 @@ const actions = {
       mode,
     }
   },
+  addLanguage(languageId, evaluate, displayName, codeMirrorName, keybinding) {
+    return {
+      type: 'ADD_LANGUAGE',
+      languageId,
```

Comment:
why do we need `languageId` bot here and in `languageDefinition`?

Reply:
ah, surely we do not!